mbed-util/CriticalSectionLock.h

```diff
@@ -47,8 +47,7 @@ class CriticalSectionLock {
 #ifdef TARGET_NORDIC
         sd_nvic_critical_region_enter(&_state);
 #else
-        _state = __get_PRIMASK();
-        __disable_irq();
+        _state = minar::platform::pushDisableIRQState();
```

Comment:
There's https://github.com/ARMmbed/core-util/blob/master/source/mbed-util_mbed.c, which implements some platform-specific code needed by `core-util`. In a similar fashion, you can create `mbed-util_posix.c`. Does that help you?